### ISSUE TITLE
Lab 07 level 2

### ISSUE THREAD

**Issue by DavidLawCheng** [nus-cs2030/2223-s2#642]
# Summary

Description of the question / problem here.

## Description:

Hi all Im stuck at level 2, do I have to create my own map method? If so, how do I do so?

## Code Snippets

Attach code snippets (if any)

## Screenshots

Screenshots (if any)

**Comment by normantan1** [comment]
Yes you have to create your own map method for Log, it is similar to exercise 4 , in this case, u return a new log with the function applied on the t value of the current Log

**Comment by xz-tan** [comment]
I think you can refer to lecture 8, where they showed how to create a map method for a context!

**Comment by DavidLawCheng** [comment]
Okok thanks all!

**Comment by nigel-tannn** [comment]
The map method is actually pretty similar to Maybe's map method, you could try referring to that!

**Comment by tjdbryan** [comment]
^ if not, you can also refer to the recitation on functional interfaces

**Comment by haqanh** [comment]
You have to create your own map method, it is quite similar to the Maybe that we have done in Exercise 4/done in recitations!

**Comment by chocococ** [comment]
i tried referring to the java api for help and also the lect notes maybe u can try that!

**Comment by user1914** [comment]
you can take a look at lecture 8 slides, slide 8! there's a sample code there

**Comment by mkim123** [comment]
Yes, you have to create your own map method. Taking a look at the Java API and lecture notes 8 will help.

**Comment by ghost1nthemachine** [comment]
Hi, are we not allowed to use IsEmpty for this lab? If so, how should I circumvent this

**Comment by normantan1** [comment]
yes you are not allowed to use isEmpty, do refer to exercise 4 or notes its roughly the same.

**Comment by tjdbryan** [comment]
> Hi, are we not allowed to use IsEmpty for this lab? If so, how should I circumvent this

You can use Optional.ofNullable!

**Comment by youthfrost** [comment]
the map method basically takes out the value from the context and map it into another value, then create a new context using the new value
